### PR TITLE
Clouseau: Add a few more gimmicks for numbers

### PR DESCRIPTION
**Pull request by scymtym**

### CODE REVIEW

File: Apps/Clouseau/src/objects/number.lisp [hunk 1]
```diff
@@ -48,19 +78,66 @@
...
       (with-style (stream :slot-like)
         (formatting-cell (stream) (write-string "Value" stream))
         (formatting-cell (stream) (declare (ignore stream))))
       (formatting-cell (stream)
         (format stream "~:D = ~:*#x~X = ~:*#o~O = ~:*#b~B" object)))
+
+    ;; Length in bits
     (format-place-row stream object 'reader-place 'integer-length
-                      :label "Length")))
+                      :label "Length")
```
Comment (kc-) [inline Apps/Clouseau/src/objects/number.lisp]: Maybe name this Bit-length? Or possibly "Length: 3 (bits)"

File: Apps/Clouseau/src/objects/number.lisp [hunk 2]
```diff
@@ -48,19 +78,66 @@
...
+                                   (symbolicate '#:most-positive- type))))
+                      `(,(constant lower)
+                        ,(constant upper)))))
+           `(cond
+              ,@(mappend #'type '(fixnum
+                                  single-float
+                                  double-float))
+              ,(constant 'pi)))))
+    (limits)))
+
```
Comment (kc-) [inline Apps/Clouseau/src/objects/number.lisp]: A matter of taste, but for integers, the `(type fixnum 0 4611686018427387903)` could maybe benefit from being shortened (or labelled with a badge), like `(type fixnum 0 most-positive-fixnum)`

**Comment by scymtym** [comment]
> I noted two suggestions in the commit comments.

Thank you.

I changed the label from "Length" to "Length (bits)" (and added the Hamming weight, while at it).

For the type in the header, I disabled printing it if it is a cons.

**Comment by kc-** [comment]
Looks good. Only one more question: for the prime factorization, it feels a little unusual to explicitly denote the exponent for singly-repeated factors. Was this a conscious decision?

**Comment by scymtym** [comment]
> Looks good. Only one more question: for the prime factorization, it feels a little unusual to explicitly denote the exponent for singly-repeated factors. Was this a conscious decision?

Just laziness. I will change it.

**Comment by scymtym** [comment]
Fixed.

**Comment by kc-** [comment]
lgtm, thank you.

**Pull request merged**